Author: Alessio Treglia <quadrispro@ubuntu.com>
Description: Add device entry for
 Creative ZEN MicroPhoto 041E:4133
 Apple iPhone 05ac:0x1290
 Apple iPod Touch 1st Gen 05ac:1291
 Apple iPhone 3G 05ac:0x1292
 Apple iPod Touch 2nd Gen 05ac:0x1293
 Apple iPhone 3GS 05ac:0x1294
 Apple 05ac:1296
 Apple 05ac:1297
 Apple 05ac:1298
 Apple iPod Touch 3rd Gen 05ac:0x1299
 Apple iPad 05ac:129a
Origin: upstream, commit:1136
Forwarded: http://sourceforge.net/support/tracker.php?aid=3053793
Bug-Ubuntu: https://launchpad.net/bugs/537386
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=594560
---
 src/music-players.h |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- libmtp.orig/src/music-players.h
+++ libmtp/src/music-players.h
@@ -52,6 +52,7 @@
   { "Creative", 0x041e, "ZEN Micro (MTP mode)", 0x4130, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Touch (MTP mode)", 0x4131, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Dell", 0x041e, "Dell Pocket DJ (MTP mode)", 0x4132, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
+  { "Creative", 0x041e, "ZEN MicroPhoto", 0x4133, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Sleek (MTP mode)", 0x4137, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN MicroPhoto", 0x413c, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Sleek Photo", 0x413d, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
@@ -1143,6 +1144,21 @@
   { "Nextar", 0x0402, "MA715A-8R", 0x5668 },
 
   /*
+   * Apple devices, which are not MTP natively but can be made to speak MTP
+   * using PwnTunes (http://www.pwntunes.net/)
+   */
+  { "Apple", 0x05ac, "iPhone", 0x1290 },
+  { "Apple", 0x05ac, "iPod Touch 1st Gen", 0x1291 },
+  { "Apple", 0x05ac, "iPhone 3G", 0x1292 },
+  { "Apple", 0x05ac, "iPod Touch 2nd Gen", 0x1293 },
+  { "Apple", 0x05ac, "iPhone 3GS", 0x1294 },
+  { "Apple", 0x05ac, "0x1296", 0x1296 },
+  { "Apple", 0x05ac, "0x1297", 0x1297 },
+  { "Apple", 0x05ac, "0x1298", 0x1298 },
+  { "Apple", 0x05ac, "iPod Touch 3rd Gen", 0x1299 },
+  { "Apple", 0x05ac, "iPad", 0x129a },
+
+  /*
    * Other strange stuff.
    */
   { "Isabella", 0x0b20, "Her Prototype", 0xddee, DEVICE_FLAG_NONE }
